lib/blob/blobstore.c

```diff
@@ -5815,6 +5836,14 @@ bs_create_blob_cpl(spdk_bs_sequence_t *seq, void *cb_arg, int bserrno)
 		spdk_spin_unlock(&blob->bs->used_lock);
 	}
 
+	/* Deallocate blob clusters in case blob creation failed.
+	 * Take into account also I/O errors that occurred during previous sequence steps.
+	 */
+	if (bserrno != 0 || seq->bserrno != 0) {
```

Comment:
Do We need to check for previous sequence error ? seq error is ultimately updated in `bs_sequence_finish` function from bserrno.

Reply:
Yes, we do.
When debugging this bug I noticed that I/O failure was passed via sequence status.